samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -0,0 +1,310 @@
+        val += (in_acc[in_id_m][in_id_n] * fil_acc[fil_m][fil_n]);
+      }
+    }
+    out_acc[id_m][id_n] = val / fil_size.size();
+  }
+};
+
+// round up function to construct the global and local size
+template <typename index_t>
+inline index_t round_up(const index_t x, const index_t y) {
+  return ((x + y - 1) / y) * y;
+}
+
+// calculating halo around each tile
+template <typename index_t>
+void inline compute_index(const index_t total_size_dim,
+                          const index_t mat_size_dim,
+                          const index_t fil_size_dim,
+                          const index_t tile_offset_dim, index_t &range_src_dim,
+                          index_t &offset_src_dim) {
+  if (tile_offset_dim == 0 && mat_size_dim < total_size_dim) {
+    offset_src_dim = tile_offset_dim;
+    range_src_dim = mat_size_dim + (fil_size_dim / 2);
+  } else if (tile_offset_dim != 0 &&
+             (tile_offset_dim + mat_size_dim) < total_size_dim) {
+    offset_src_dim = tile_offset_dim - (fil_size_dim / 2);
+    range_src_dim = mat_size_dim + fil_size_dim - 1;
+
+  } else if (tile_offset_dim != 0 &&
+             (tile_offset_dim + mat_size_dim) >= total_size_dim) {
+    offset_src_dim = tile_offset_dim - (fil_size_dim / 2);
+    range_src_dim = mat_size_dim + (fil_size_dim / 2);
+  } else if (tile_offset_dim == 0 && mat_size_dim >= total_size_dim) {
+    offset_src_dim = tile_offset_dim;
+    range_src_dim = mat_size_dim;
+  }
+}
+
+int main() {
+  using data_t = float;
+  using index_t = int;
+  // sycl read type
+  static constexpr auto read_t = cl::sycl::access::mode::read;
+  // sycl write type
+  static constexpr auto write_t = cl::sycl::access::mode::write;
+  // sycl global buffer type
+  static constexpr auto global_buffer_t =
+      cl::sycl::access::target::global_buffer;
+  // read_accessor type
+  using read_accessor_t =
+      cl::sycl::accessor<data_t, 2, read_t, global_buffer_t>;
+  // write accessor type
+  using write_accessor_t =
+      cl::sycl::accessor<data_t, 2, write_t, global_buffer_t>;
+  // opencl configuration type
+  using ocl_config_t = opencl_configuration_t<index_t>;
+
+  // total input data size
+  auto total_buffer = matrix_size_t<index_t>{1024, 1024};
+  // tile size per iteration
+  auto mat_size = matrix_size_t<index_t>{512, 512};
+  auto fil_size = matrix_size_t<index_t>{3, 3};
+
+  // constructing the tile size
+  auto num_host_tile_n = total_buffer.n / mat_size.n;
+  auto num_host_tile_m = total_buffer.m / mat_size.m;
+  // input value
+  auto input_data = data_t(0.6);
+  // mask filter value
+  auto filter_data = data_t(0.3);
+  // input array
+  std::vector<data_t> input(total_buffer.size(), input_data);
+  // mask array
+  std::vector<data_t> filter(fil_size.size(), filter_data);
+
+  auto property_list =
+      cl::sycl::property_list{cl::sycl::property::queue::enable_profiling()};
+  // constructing a SYCL queue for OpenCL device where
```

Comment:
I don't think this comment is useful either - as a longer sample, I think we should be teaching about the algorithms. We have some more basic samples that talk about queues and what they do.